### PR TITLE
Fix WithBaseCall and WithThisCall using parameter type instead of name

### PR DESCRIPTION
**Pull request by zsr2531**
I also made the "unsafe" overloads more flexible, now you can pass constants like `123` or `"hello world"` to base/this.

### CODE REVIEW

**Comment by zsr2531** [comment]
Closes #4

**Comment by zsr2531** [comment]
Ready to merge, if you have no issues/suggestions. :rocket: @dansiegel

File: src/CodeGenHelpers/ConstructorBuilder.cs [hunk 1]
```diff
@@ -144,25 +134,15 @@ public ConstructorBuilder WithThisCall(IMethodSymbol baseConstructor)
...
 
-                var output = _parameters.Where(x => baseConstructor.Parameters.Any(p => p.Type.Name == x.Key || p.Type.GetFullMetadataName() == x.Key)).Select(x => x.Value);
+                var output = _parameters.Where(x => baseConstructor.Parameters.Any(p => p.Type.Name == x.Key || p.Type.GetFullMetadataName() == x.Key)).Select(x => x.Key);
                 return $": this({string.Join(", ", output)})";
             };
             return this;
         }
 
-        public ConstructorBuilder WithBaseCall(Dictionary<string, string> parameters)
+        public ConstructorBuilder WithBaseCall(params string[] parameters)
```
Comment (dansiegel) [inline src/CodeGenHelpers/ConstructorBuilder.cs:143]: I'd rather that you introduce an override than change the signature outright. You're also outright breaking expected behavior here where the base parameters should be checking for and being added to the parameters in the main constructor... so you might have:

```cs
// expected
public class Foo : Bar
{
    public Foo(string someValue, Baz baz) : base(baz)
    {
    }
}

// builder
constructorBuilder
    .AddParameter("string", "someValue")
    .WithBaseCall(new Dictionary<string, string> { { "Baz", "baz" } });
```

**Review by dansiegel — CHANGES_REQUESTED** [review]

**Comment by dansiegel** [comment]
I'm going to close this PR... there are some other enhancements I need to make around providing default values for parameters. I'll address this issue as part of that PR.

**Pull request closed without merging**